scripts/data/client.py

```diff
@@ -56,41 +64,44 @@ def __str__(self):
         return f"Job(height='{self.height}', step={self.step}, weight='{self.weight}')"
 
 
-# Generator function to create jobs
+def calculate_batch_weight(block_data, mode):
+    if mode == "light":
+        return len(block_data["blocks"])
+    else:
+        return sum(
+            len(tx["inputs"]) + len(tx["outputs"])
+            for block in block_data["blocks"]
+            for tx in block["data"]["transactions"]
+        )
+
+
 def job_generator(start, blocks, step, mode, strategy):
     BASE_DIR.mkdir(exist_ok=True)
     end = start + blocks
-
     height_range, step = (
         ([random.randint(start, end) for _ in range(start, end)], 1)
         if strategy == "random"
         else (range(start, end, step), step)
     )
-
     for height in height_range:
+        if shutdown_event.is_set():
```

Comment:
No need to check `shutdown_event` here, it is checked in `job_producer`:
![image](https://github.com/user-attachments/assets/926b9a7b-2482-44e1-a427-51cf806980fd)

Reply:
Why this is resolved? job_generator and process_batch is about bussines logic those functions should not bother with shutdown subtleties.